lib/src/code_generator/imports.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Represents a library import which will be written as an import in the
+/// generated file.
+class LibraryImport {
+  final String name;
+  final String importPath;
+  late String prefix;
+
+  LibraryImport(this.name, this.importPath) {
+    prefix = name;
+  }
+
+  @override
+  bool operator ==(other) {
+    return other is LibraryImport && name == other.name;
+  }
+
+  @override
+  int get hashCode => name.hashCode;
+}
+
+/// Represents an imported type.
+class ImportedType {
+  final LibraryImport libraryImport;
+  final String cType;
+  final String dartType;
+
+  ImportedType(this.libraryImport, this.cType, this.dartType);
+}
+
+final ffiImport = LibraryImport('ffi', 'dart:ffi');
+final ffiPkgImport = LibraryImport('pkg_ffi', 'package:ffi/ffi.dart');
+
+final voidType = ImportedType(ffiImport, 'Void', 'void');
+
+final unsignedCharType = ImportedType(ffiImport, 'Uint8', 'int');
+final charType = ImportedType(ffiImport, 'Int8', 'int');
+final unsignedShortType = ImportedType(ffiImport, 'Uint16', 'int');
+final shortType = ImportedType(ffiImport, 'Int16', 'int');
+final unsignedIntType = ImportedType(ffiImport, 'Uint32', 'int');
+final intType = ImportedType(ffiImport, 'Int32', 'int');
+final unsignedLongType = ImportedType(ffiImport, 'Uint64', 'int');
```

Comment:
This variable name is incorrect.

`(unsigned) long` is not (always) `(u)int64_t`. https://dart-review.googlesource.com/c/sdk/+/221501/28/tests/ffi/abi_specific_ints.dart#76

You can play around with https://godbolt.org/z/ee8MTjY8Y if you want to inspect the sizes. You can select the compiler (win/linux x64/x86) in the dropdown and see the size as an int in the generated assembly.

Reply:
Updated to use types from package:ffi.